### ISSUE TITLE
rlof_mask and pessimistic_mask in setCOMPASDCOmask() of classCOMPAS not being set correctly?

### ISSUE THREAD

**Issue by jeffriley** [TeamCOMPAS/COMPAS#726]
**Describe the bug**
I think the masks rlof_mask and pessimistic_mask in setCOMPASDCOmask() of class COMPAS (ClassCOMPAS.py) are not being set correctly (because rlof_flag and pessimistic_flag are not treated as boolean), resulting in an incorrect count for the selected DCOs ("BBH", "BNS", etc.).

A while ago, @LiekeVanSon (I think) had an issue with hubble_mask in the same function, and found that the boolean columns from the COMPAS HDF5 files were not being treated as booleans by Python - @LiekeVanSon solved this by adding ".astype(bool)' to hubble_flag when setting hubble_mask.  I think the same is necessary for rlof_flag and pessimistic_flag in the following code:

            # if masking on RLOF, get flag and match seeds to dco seeds
            if noRLOFafterCEE:
                rlof_flag = self.get_COMPAS_variables("BSE_Common_Envelopes", "Immediate_RLOF>CE")[dco_from_ce].astype(bool)
                rlof_seeds = np.unique(dco_ce_seeds[rlof_flag])
                rlof_mask = np.logical_not(np.in1d(dco_seeds, rlof_seeds))
            else:
                rlof_mask = np.repeat(True, len(dco_seeds))

            # if masking on pessimistic CE, get flag and match seeds to dco seeds
            if pessimistic:
                pessimistic_flag = self.get_COMPAS_variables("BSE_Common_Envelopes", "Optimistic_CE")[dco_from_ce].astype(bool)
                pessimistic_seeds = np.unique(dco_ce_seeds[pessimistic_flag])
                pessimistic_mask = np.logical_not(np.in1d(dco_seeds, pessimistic_seeds))
            else:
                pessimistic_mask = np.repeat(True, len(dco_seeds))

Suggested changes are:

                rlof_flag = (self.get_COMPAS_variables("BSE_Common_Envelopes", "Immediate_RLOF>CE")[dco_from_ce].astype(bool)).astype(bool)

                pessimistic_flag = (self.get_COMPAS_variables("BSE_Common_Envelopes", "Optimistic_CE")[dco_from_ce].astype(bool)).astype(bool)

**Label the issue**
`urgency_moderate` - This is a moderately urgent issue
`severity_moderate` - This is a moderately severe bug

**To Reproduce**
Run COMPAS to get a population of binaries, then run the PostProcessing code - FastCosmicIntegation.py - I believe the number of selected DCOs ("BBH", "BNS", etc.) is wrong because of this defect.

**Expected behavior**
The count of selected DCOs is correct.

**Screenshots**

**Versioning (please complete the following information):**
 - OS: Ubuntu 20.04
 - COMPAS v02.26.01

**Additional context**

**Comment by jeffriley** [comment]
Maybe it would be a good idea to change get_COMPAS_variables() to check the data type, and if necessary add the '.astype(bool)' there.  I know get_COMPAS_variables() is not enforced or always used, but it might help, and if we document what we're doing and why there, it might help others in  the future.

Take a look at printHeaders() and getDataType() in h5view.py to see how to get the data type from the HDF5 file.

**Comment by LiekeVanSon** [comment]
Hi Jeff, much agreed! I think your proposed solution should be fine.

**Comment by FloorBroekgaarden** [comment]
taking a look at this 👀

**Comment by FloorBroekgaarden** [comment]
I took a thorough look at this and tested all the functionality of the DCOmask with COMPAS test data. 
there is no issue as the other masks (ImmidiateRLOF and Optimistic CE) are already set to bool type with the `.astype(bool)` in the lines 

`rlof_flag = self.get_COMPAS_variables("BSE_Common_Envelopes", "Immediate_RLOF>CE")[dco_from_ce].astype(bool)`
`self.get_COMPAS_variables("BSE_Common_Envelopes", "Optimistic_CE")[dco_from_ce].astype(bool)`

no extra .astype(bool) is required. 

I have added tiny line of documentation for future developers that are adding the mask function in this python script. 

Issue can be closed, and what is left is addressed with PR #895
